### PR TITLE
topology: sof-tgl-max98357a-rt5682: support rt1011 amp

### PR DESCRIPTION
**Pull request by macchian**
Add a macro 'CODEC' to generate topology for rt1011 speaker amp.
Add a macro 'FMT' to seperate speaker dai format: s16le, s24le

Signed-off-by: Brent Lu <brent.lu@intel.com>
Signed-off-by: Mac Chiang <mac.chiang@intel.com>

### CODE REVIEW

File: tools/topology/sof-tgl-max98357a-rt5682.m4 [hunk 1]
```diff
@@ -231,7 +231,7 @@ DAI_ADD(sof/pipe-dai-playback.m4,
 
 # PCM Low Latency, id 0
 dnl PCM_PLAYBACK_ADD(name, pcm_id, playback)
-PCM_PLAYBACK_ADD(max357a-spk, 0, PIPELINE_PCM_1)
+PCM_PLAYBACK_ADD(CODEC, 0, PIPELINE_PCM_1)
```
Comment (plbossart) [inline tools/topology/sof-tgl-max98357a-rt5682.m4:234]: this should be 'Speakers'

**Review by plbossart — CHANGES_REQUESTED** [review]
minor change needed for user-visible string.

**Comment by brentlu** [comment]
PR of machine driver: https://github.com/thesofproject/linux/pull/2586

File: tools/topology/sof-tgl-max98357a-rt5682.m4 (hunk 1 above)
Comment (macchian) [reply tools/topology/sof-tgl-max98357a-rt5682.m4]: @plbossart , Thanks for pointing out! I commit new one for your reviewing.

File: tools/topology/sof-tgl-max98357a-rt5682.m4 [hunk 2]
```diff
@@ -69,7 +69,10 @@ MUXDEMUX_CONFIG(demux_priv_1, 2, LIST(`	', `matrix1,', `matrix2'))
 # PCM99 <---- volume <---- DMIC01 (dmic 48k capture)
 # PCM100 <---- kpb <---- DMIC16K (dmic 16k capture)
 
-# Define pipeline id for sof-tgl-max98357a-rt5682.m4
+# Define pipeline id for sof-tgl-CODEC-rt5682.m4
+ifelse(CODEC, `MAX98357A', `define(SPK_PCM_NAME, `max357a-spk')')
+ifelse(CODEC, `RT1011', `define(SPK_PCM_NAME, `Speakers')')
```
Comment (plbossart) [inline tools/topology/sof-tgl-max98357a-rt5682.m4:74]: why not make them 'Speakers' for both? that should have no impact on userspace/UCM?

**Review by plbossart — CHANGES_REQUESTED** [review]

File: tools/topology/sof-tgl-max98357a-rt5682.m4 (hunk 2 above)
Comment (macchian) [reply tools/topology/sof-tgl-max98357a-rt5682.m4]: @plbossart , You are right! No impact. So fix the string 'Speakers' for both.

**Comment by xiulipan** [comment]
SOFCI TEST

**Review by plbossart — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
Jenkins showing unrelated kernel PM issue on ICL

**Comment by lgirdwood** [comment]
Travis CI timeout.

**Pull request merged**